app/src/androidTest/java/com/github/h3lp3rs/h3lp/HelpeeSelectionActivityTest.kt

```diff
@@ -0,0 +1,212 @@
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.intent.Intents.*
+import androidx.test.espresso.intent.matcher.IntentMatchers.*
+import androidx.test.espresso.matcher.RootMatchers
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.GrantPermissionRule
+import com.github.h3lp3rs.h3lp.database.Databases.Companion.setDatabase
+import com.github.h3lp3rs.h3lp.database.Databases.EMERGENCIES
+import com.github.h3lp3rs.h3lp.database.Databases.NEW_EMERGENCIES
+import com.github.h3lp3rs.h3lp.database.MockDatabase
+import com.github.h3lp3rs.h3lp.dataclasses.MedicalInformation
+import com.github.h3lp3rs.h3lp.locationmanager.GeneralLocationManager
+import com.github.h3lp3rs.h3lp.signin.SignInActivity.Companion.globalContext
+import com.github.h3lp3rs.h3lp.signin.SignInActivity.Companion.userUid
+import com.github.h3lp3rs.h3lp.storage.Storages
+import com.github.h3lp3rs.h3lp.storage.Storages.Companion.resetStorage
+import com.github.h3lp3rs.h3lp.storage.Storages.Companion.storageOf
+import com.github.h3lp3rs.h3lp.storage.Storages.MEDICAL_INFO
+import org.hamcrest.Matchers.*
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+// Case example of a possible query when a user clicks on the call for emergency button
+@RunWith(AndroidJUnit4::class)
+class HelpParametersActivityTest : H3lpAppTest() {
+
+    @get:Rule
+    val testRule = ActivityScenarioRule(
+        HelpeeSelectionActivity::class.java
+    )
+
+    @get:Rule
+    var mRuntimePermissionRule: GrantPermissionRule =
+        GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION)
+
+    @Before
+    fun setUp() {
+       // initIntentAndCheckResponse()
+
+        globalContext = getApplicationContext()
+        userUid = USER_TEST_ID
+
+        setDatabase(NEW_EMERGENCIES, MockDatabase())
+
+        val emergencyDb = MockDatabase()
+        emergencyDb.setInt(globalContext.getString(R.string.EMERGENCY_UID_KEY), 0)
+        setDatabase(EMERGENCIES, emergencyDb)
+
+        resetStorage()
+
+        storageOf(Storages.USER_COOKIE).setBoolean(
+            globalContext.getString(R.string.KEY_USER_AGREE),
+            true
+        )
+    }
+
+    @Test
+    fun clickSearchHelpWithMedsWorksAndSendsIntent() {
+        launchAndDo {
+            // select one med
+            val medButton0 = onView(withId(R.id.selectMedsButton0))
+            medButton0.check(matches(isNotChecked()))
+
+            // select the medication
+            medButton0.perform(click())
+
+            // click the search help button
+            val searchHelpButton = onView(withId(R.id.help_params_search_button))
+            searchHelpButton.check(matches(isDisplayed()))
+            searchHelpButton.perform(click())
```

Comment:
Not very important, but this could be a separate private method, since it is used twice